Source/WebKit/UIProcess/API/glib/WebKitImage.cpp

```diff
@@ -0,0 +1,476 @@
+
+    g_object_class_install_properties(objectClass, N_PROPERTIES, sObjProperties.data());
+}
+
+// When instantiating this class and providing a CompletionHandler to instantiate the bytes, make sure
+// this uses #CompletionHandlerCallThread::AnyThread if you first instantiate the class bytes asynchronously.
+WebKitImage* webkitImageNew(int width, int height, guint stride, CompletionHandler<GRefPtr<GBytes>(WebKitImage*)>&& createBytesFn)
+{
+    RELEASE_ASSERT(createBytesFn);
+    RELEASE_ASSERT(static_cast<gsize>(width) * RGBA8BytesPerPixel <= stride);
+
+    auto* image = WEBKIT_IMAGE(g_object_new(WEBKIT_TYPE_IMAGE,
+        "width", width,
+        "height", height,
+        "stride", stride,
+        nullptr));
+    image->priv->createBytesFn = WTFMove(createBytesFn);
+
+    return image;
+}
+
+/**
+ * webkit_image_get_width:
+ * @image: a #WebKitImage
+ *
+ * Get the @image width in pixels
+ *
+ * Returns: the image width
+ *
+ * Since: 2.52
+ */
+int webkit_image_get_width(WebKitImage* image)
+{
+    g_return_val_if_fail(WEBKIT_IS_IMAGE(image), 0);
+    return image->priv->width;
+}
+
+/**
+ * webkit_image_get_height:
+ * @image: a #WebKitImage
+ *
+ * Get the @image height in pixels
+ *
+ * Returns: the image height
+ *
+ * Since: 2.52
+ */
+int webkit_image_get_height(WebKitImage* image)
+{
+    g_return_val_if_fail(WEBKIT_IS_IMAGE(image), 0);
+    return image->priv->height;
+}
+
+/**
+ * webkit_image_get_stride:
+ * @image: a #WebKitImage
+ *
+ * Get the @image stride
+ *
+ * Returns: the image stride
+ *
+ * Since: 2.52
+ */
+guint webkit_image_get_stride(WebKitImage* image)
+{
+    g_return_val_if_fail(WEBKIT_IS_IMAGE(image), 0);
+    return image->priv->stride;
+}
+
+/**
+ * webkit_image_as_rgba_bytes:
+ * @image: a #WebKitImage
+ *
+ * Get the @image pixel data as an array of bytes.
+ *
+ * The pixel format for the returned byte buffer is 32-bit per pixel
+ * with 8-bit premultiplied alpha, in the preferred byte order for
+ * the architecture (typically ABGR8888 on little-endian hosts, and
+ * RGBA8888 on big-endian ones).
+ *
+ * Returns: (transfer none): a #GBytes
+ *
+ * Since: 2.52
+ */
+GBytes* webkit_image_as_rgba_bytes(WebKitImage* image)
+{
+    g_return_val_if_fail(WEBKIT_IS_IMAGE(image), nullptr);
+
+    if (!image->priv->bytes) {
+        ASSERT(image->priv->createBytesFn);
+        image->priv->bytes = image->priv->createBytesFn(image);
+    }
+    ASSERT(image->priv->bytes);
+
+    return image->priv->bytes.get();
+}
+
+#if PLATFORM(GTK)
+/**
+ * webkit_image_to_gdk_texture:
+ * @image: a #WebKitImage
+ *
+ * Get the @image as a #GdkTexture
+ *
+ * Returns: (transfer full): a #GdkTexture
+ *
+ * Since: 2.52
+ */
+GdkTexture* webkit_image_to_gdk_texture(WebKitImage *image)
+{
+    g_return_val_if_fail(WEBKIT_IS_IMAGE(image), nullptr);
+
+    auto* bytes = webkit_image_as_rgba_bytes(image);
+
+    return gdk_memory_texture_new(image->priv->width, image->priv->height, GDK_MEMORY_DEFAULT, bytes, image->priv->stride);
```

Comment:
I'm not sure it's worth exposing this, since the caller can just use gdk_memory_texture_new